build: adjust the installer build and artifact staging

compnerd opened the pull request:
Adjust the installer to allow it to be built for foreign architectures. Additionally, adjust the artifact staging to include additional content that was missed previously in order to enable code signing.

compnerd commented:
CC: @tristanlabelle @shahmishal

compnerd commented:
@swift-ci please build toolchain Windows platform

tristanlabelle commented on utils/build.ps1:
```diff
@@ -1655,9 +1661,11 @@ if (-not $SkipBuild) {
+  Build-Installer $HostArch
 }
 
+Build-ArtifactPresentation $HostARch
```
CAsing

tristanlabelle commented on utils/build.ps1:
```diff
@@ -1546,46 +1546,52 @@ function Build-DocC() {
-    # Extract installer engine to ease code-signing on swift.org CI
-    New-Item -Type Directory -Path "$($HostArch.BinaryCache)\installer\$($HostArch.VSName)\" -ErrorAction Ignore | Out-Null
-    Invoke-Program "$BinaryCache\wix-4.0.1\tools\net6.0\any\wix.exe" -- burn detach "$($HostArch.BinaryCache)\installer\Release\$($HostArch.VSName)\installer.exe" -engine "$Stage\installer-engine.exe" -intermediateFolder "$($HostArch.BinaryCache)\installer\$($HostArch.VSName)\"
+function Build-ArtifactPresentation($Arch) {
```
It's a bit of a stretch to call this a `Build` operation. Maybe, `Stage-InstallerArtifacts`?

compnerd replied:
Fair

compnerd commented:
@swift-ci please build toolchain Windows platform

compnerd committed:
build: adjust the installer build and artifact staging

Adjust the installer to allow it to be built for foreign architectures.
Additionally, adjust the artifact staging to include additional content
that was missed previously in order to enable code signing.

compnerd commented:
@swift-ci please build toolchain Windows platform

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please build toolchain Windows platform

compnerd commented:
@swift-ci please build toolchain Windows platform

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please test Windows platform
